cycler.py

```diff
@@ -124,6 +124,43 @@ def keys(self):
         """
         return set(self._keys)
 
+    def change_key(self, old, new):
+        """
+        Change a key in this cycler to a new name.
+        Modification is performed in-place.
+
+        Does nothing if the old key is the same as the new key.
+        Raises a ValueError if the new key is already a key.
+        Raises a KeyError if the old key isn't a key.
+
+        """
+        if old == new:
+            return
+        if new in self._keys:
+            raise ValueError("Can't replace %s with %s, %s is already a key" %
+                             (old, new, new))
+        if old not in self._keys:
+            raise KeyError("Can't replace %s with %s, %s is not a key" %
+                           (old, new, old))
+
+        self._keys.remove(old)
+        self._keys.add(new)
+
+        if self._right is not None and old in self._right.keys:
+            self._right.change_key(old, new)
+
+        # self._left should always be non-None
+        # if self._keys is non-empty.
+        elif isinstance(self._left, Cycler):
+            self._left.change_key(old, new)
+        else:
+            # It should be completely safe at this point to
+            # assume that the old key can be found in each
+            # iteration.
+            for entry in self._left:
+                entry[new] = entry[old]
```

Comment:
Ok, so it looks like the `cycler(newlabel, oldcycler)` trick is broken.
```
>>> c = cycler('lw', range(3))
>>> cc = cycler('linewidth', c)
>>> cc.keys
set(['linewidth'])
>>> for v in cc: print(v)
{'lw': 1}
{'lw': 2}
{'lw': 3}
```
I'll take a look at this.